### PR TITLE
Adding missing operation wrappers and syntax kinds to the shim layer

### PR DESCRIPTION
**Pull request by Tim-Pohlmann**

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/OperationInterfaces.xml [hunk 1]
```diff
@@ -3036,4 +3122,273 @@
...
+    <Property Name="AppendCall" Type="IOperation">
+      <Comments>
+        <summary>
+          If this interpolated string is subject to an interpolated string handler conversion, the construction of the interpolated string handler instance.
+          This can be an <see cref="IInvocationOperation"/>  or <see cref="IDynamicInvocationOperation"/> for valid code, and <see cref="IInvalidOperation"/> for invalid code.
+        </summary>
+      </Comments>
+    </Property>
+  </Node>
+  <!--<Node Name="IInterpolatedStringHandlerArgumentPlaceholderOperation" Base="IOperation">  Commented out because it is not needed and InterpolatedStringArgumentPlaceholderKind is missing.
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/ShimLayer/OperationInterfaces.xml:3210]: Add the excluded type "InterpolatedStringArgumentPlaceholderKind" here instead:
```cs
                this.IsSkipped = this.Type switch
                {
                    "ArgumentKind" => true,
                    "BranchKind" => true,
                    "CaseKind" => true,
                    "CommonConversion" => true,
                    "ForEachLoopOperationInfo" => true,
                    "IDiscardSymbol" => true,
                    "InstanceReferenceKind" => true,
                    "LoopKind" => true,
                    "PlaceholderKind" => true,
                    _ => !this.IsPublicProperty,
                };
```

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/OperationInterfaces.xml [hunk 2]
```diff
@@ -827,6 +858,7 @@
         <summary>Referenced property.</summary>
       </Comments>
     </Property>
+    <!--<Property Name="ConstrainedToType" Type="ITypeSymbol?" Override="true"/> Commented out to avoid duplication-->
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/ShimLayer/OperationInterfaces.xml:861]: Please mark comments by us with "Sonar:" to distinguish them from the Roslyn comments.

Consider adding simple support for "Override" by excluding them in
```cs
                this.IsSkipped = this.Type switch
                {
                    "ArgumentKind" => true,
                    "BranchKind" => true,
                    "CaseKind" => true,
                    "CommonConversion" => true,
                    "ForEachLoopOperationInfo" => true,
                    "IDiscardSymbol" => true,
                    "InstanceReferenceKind" => true,
                    "LoopKind" => true,
                    "PlaceholderKind" => true,
                    _ => !this.IsPublicProperty && !this.IsOverride, // <- IsOverride needs to be initiated in the ctor.
                };
```

If you do so, we need an issue tracking the work to include support for overrides.

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/OperationInterfaces.xml (hunk 2 above)
Comment (Tim-Pohlmann) [reply analyzers/src/SonarAnalyzer.CFG/ShimLayer/OperationInterfaces.xml]: #8283

File: analyzers/src/SonarAnalyzer.ShimLayer.CodeGeneration/OperationLightupGenerator.cs [hunk 3]
```diff
@@ -1069,6 +1071,8 @@ public PropertyData(XElement node)
 
             public bool IsPublicProperty { get; }
 
+            public bool IsOverride{ get; }
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.ShimLayer.CodeGeneration/OperationLightupGenerator.cs:1074]:
```suggestion
            public bool IsOverride{ get; } // Added by Sonar. Usages are also by Sonar.
```

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/SyntaxKindEx.cs [hunk 4]
```diff
@@ -67,6 +67,7 @@ public static class SyntaxKindEx
         public const SyntaxKind OrPattern = (SyntaxKind)9031;
         public const SyntaxKind AndPattern = (SyntaxKind)9032;
         public const SyntaxKind NotPattern = (SyntaxKind)9033;
+        public const SyntaxKind SlicePattern = (SyntaxKind)9034;
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/ShimLayer/SyntaxKindEx.cs:70]: Educational: Changes in this file seem to be unrelated. Are they necessary for the PR? If so, how?

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/OperationInterfaces.xml [hunk 5]
```diff
@@ -127,7 +127,7 @@
         </para>
       </summary>
     </Comments>
-    <Property Name="LoopKind" Type="LoopKind">
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.CFG/ShimLayer/OperationInterfaces.xml:130]: Is this file copied unmodified now?

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM. But please anwser the questions and consider adding the suggestion.

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/SyntaxKindEx.cs (hunk 4 above)
Comment (Tim-Pohlmann) [reply analyzers/src/SonarAnalyzer.CFG/ShimLayer/SyntaxKindEx.cs:70]: `SlicePattern` is needed in the Security repo. This could be a separate PR, but I thought it would be easier to include it here (thematically fitting) instead of creating a 1-line PR.

File: analyzers/src/SonarAnalyzer.CFG/ShimLayer/OperationInterfaces.xml (hunk 5 above)
Comment (Tim-Pohlmann) [reply analyzers/src/SonarAnalyzer.CFG/ShimLayer/OperationInterfaces.xml:130]: The only difference is the lack of `?` in our version of the file. The rest is identical (including formatting inconsistencies) to make maintenance easier.
